principles/code_is_not_yours.md

```diff
@@ -0,0 +1,74 @@
+
+There are things at our disposal to help with this.
+
+## Good naming
+
+Naming is hard.
+
+> There are only two hard things in Computer Science: cache invalidation and naming things.
+>
+> -- Phil Karlton
+
+That said you can avoid basic errors such as
+
+- Generic variable names: `p`, `temp`, `v`
+- Using arrays over hashes: `['Bob', 'Smith', 'Bristol']` over `{ name: 'Bob', surname: 'Smith', location: 'Bristol' }`
+- Use of acronyms: particularly domain specific ones. Someone new to the service may not know their GOR from their WIMS
+
+Other than that when considering the name for a method, class, module etc try not to approach it from what makes sense to you, but instead what best describes the intent of the thing, and what will help the person who comes after you.
+
+## Relevant documentation
+
+Lets be frank; few developers like writing documentation, whether its comments in the code or actual documentation like the README or accompanying wiki. So this principle doesn't advocate that **everything** should be commented. There is little value in documenting that `Person.name` gets and sets a person's name!
+
+### When writing code document the *why*
+
+Don't document what a thing is doing, that should be clear from the code. Document *why* the thing is needed, or *why* its doing it the way it is.
+
+### When documenting the project document the *how*
+
+When it comes to other developers or users of the project, the thing they are most interested in is the *how*. *How* can I build/deploy/run this project. *How* can I use it in my own work.
+
+## Clean commit history
+
+A clean commit history with meaningful messages is essential to understanding the intent of code and why its there.
+
+The [Pull requests](/process/pull_request.md) process was put together to support this. Stick to the following and your commit history should take care of itself.
+
+- **Never commit to master** always make changes on a branch
+- **Rebase your branch on master** instead of merge when keeping it up to date
+- **Merge with `--no-ff`** when finished if you have to merge locally. This retains the fact the branch existed. Ideally though you're creating pull requests in GitHub and using it to merge so this will be done for you.
+- **Record *why* the commit was made** over the what the commit is doing (though a summary of changes is nice!)
+
+Finally using the following commit message template will help with some of this, and keep things consistent.
+
+```text
+50 character limited title
+
+Link to originating story/bug/card in relevant system (e.g. Jira or Trello)
+
+Brief description covering why we're making this change.
+```
+
+For example
+
+```text
+Handle empty params in ValidatePreUpdate method
+
+https://eaflood.atlassian.net/browse/WAS-1096
+
+This change fixes an issue found with `validate_pre_update_organisation_address`
+in that when passed empty parameters it would cause an `undefined method all?
+for nil:NilClass` error to be thrown.
+```
```

Comment:
Think I was just trying to hammer home the message, but think I agree with you in hindsight. Watch this space!